### PR TITLE
feat: add stableswap pallet

### PR DESCRIPTION
**Pull request by martinfridrich**
This PR adds stableswap pallel from warehouse into the runtime.

todo:

- [x] benchmarks
- [ ] pallet's runtime configuration (amplification, trading limits, authority origin)

### CODE REVIEW

File: integration-tests/src/router/different_pools.rs [hunk 1]
```diff
@@ -0,0 +1,138 @@
+#![cfg(test)]
+#![allow(clippy::identity_op)]
+use crate::assert_trader_bsx_balance;
+use crate::assert_trader_non_native_balance;
+
+use super::*;
+
+#[test]
+fn sell_should_work_when_route_contains_trades_with_different_pools() {
```
Comment (dmoka) [inline integration-tests/src/router/different_pools.rs:9]: Love the separation of the router tests, great job!

File: runtime/basilisk/src/lib.rs [hunk 2]
```diff
@@ -928,10 +929,27 @@ impl pallet_route_executor::Config for Runtime {
...
+	type Event = Event;
+	type AssetId = AssetId;
+	type Currency = Currencies;
+	type ShareAccountId = AssetsAccountId<Self>;
+	type AssetRegistry = AssetRegistry;
+	type AuthorityOrigin = SuperMajorityTechCommitteeOrRoot;
+	type MinPoolLiquidity = StableswapMinPoolLiquidity;
+	type MinTradingLimit = StableswapMinTradingLimit;
+	type AmplificationRange = AmplificationRange;
+	type WeightInfo = (); //TODO:
```
Comment (dmoka) [inline runtime/basilisk/src/lib.rs:950]: todo

File: runtime/testing-basilisk/src/lib.rs [hunk 3]
```diff
@@ -893,10 +893,23 @@ impl pallet_route_executor::Config for Runtime {
...
+	type Event = Event;
+	type AssetId = AssetId;
+	type Currency = Currencies;
+	type ShareAccountId = AssetsAccountId<Self>;
+	type AssetRegistry = AssetRegistry;
+	type AuthorityOrigin = SuperMajorityTechCommitteeOrRoot;
+	type MinPoolLiquidity = StableswapMinPoolLiquidity;
+	type MinTradingLimit = StableswapMinTradingLimit;
+	type AmplificationRange = AmplificationRange;
+	type WeightInfo = (); //TODO:
```
Comment (dmoka) [inline runtime/testing-basilisk/src/lib.rs:910]: todo

Comment (martinfridrich) [reply runtime/testing-basilisk/src/lib.rs:910]: yes, this is only draft. There are missing benchmarks in the warehouse. I'll implement it and fix these but I'm working on something else for now.

**Pull request closed without merging**